src/components/Swap/Speedbump/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { Trans } from '@lingui/macro'
+import Button, { TextButton } from 'components/Button'
+import Column from 'components/Column'
+import { OnCloseDialogContext } from 'components/Dialog'
+import Row from 'components/Row'
+import { PriceImpact } from 'hooks/usePriceImpact'
+import { LargeAlert, LargeIcon, X } from 'icons'
+import { PropsWithChildren, useContext } from 'react'
+import styled, { css } from 'styled-components/macro'
+import { ThemedText } from 'theme'
+
+const SpeedbumpWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  padding: 1em;
+  text-align: center;
+`
+
+const Body = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  justify-content: space-between;
+`
+
+const BodyText = styled(ThemedText.Body1)`
+  padding: 0 0.5em;
+`
+
+const IconWrapper = styled.div`
+  padding: 2em;
+`
+const SpeedbumpButtonStyle = css`
+  border-radius: 1em;
+  padding: 1em;
+`
+const HeaderRow = styled(Row)`
+  align-items: flex-start;
+  width: 100%;
+`
+
+const StyledXButton = styled(LargeIcon).attrs({ icon: X, color: 'primary', size: 1.5 })`
+  :hover {
+    cursor: pointer;
+  }
+`
+const ContinueButton = styled(Button)`
+  ${SpeedbumpButtonStyle}
+  background-color: ${({ theme }) => theme.criticalSoft};
+  color: ${({ theme }) => theme.critical};
+`
+const CancelButton = styled(TextButton)`
+  ${SpeedbumpButtonStyle}
+  color: ${({ theme }) => theme.secondary};
+`
+
+interface SpeedbumpDialogProps {
+  impact: PriceImpact
+  onAcknowledge: () => void
+}
+
+export function SpeedbumpDialog({ impact, onAcknowledge, children }: PropsWithChildren<SpeedbumpDialogProps>) {
+  const onClose = useContext(OnCloseDialogContext)
```

Comment:
we tend to have a pattern of keeping `Context` themselves private, and exposing a hook for the specific thing you need, to prevent improper use of the Context. it would look like exporting this in the Dialog file:

```
export const useOnDialogClose = () =>  {
    return useContext(OnCloseDialogContext)
}
```

i don't think there's much difference in this case, and i only mention it because of the pattern i've seen elsewhere.